### PR TITLE
Feat/no staking test case

### PR DESCRIPTION
**Pull request by Mohsinsiddi**
Added  Test Script for No Staking Option

### CODE REVIEW

File: tests/test_run_service.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                raise Exception("Docker container check failed")
+                
+            # Check service health
+            if not check_service_health(logger):
+                raise Exception("Service health check failed")
+                
+            logger.info("All checks passed successfully")
+        else:
+            logger.error(f"Test failed with exit status {child.exitstatus}")
+
```
Comment (OjusWiZard) [inline tests/test_run_service.py:258]: this should fail the test, right?
```suggestion
            raise Exception(f"Test failed with exit status {child.exitstatus}")
            
```

File: tests/test_run_service.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            # Check service health
+            if not check_service_health(logger):
+                raise Exception("Service health check failed")
+                
+            logger.info("All checks passed successfully")
+        else:
+            logger.error(f"Test failed with exit status {child.exitstatus}")
+            
+    except Exception as e:
+        logger.error(f"Test failed with error: {str(e)}")
```
Comment (OjusWiZard) [inline tests/test_run_service.py:260]: this also should fail the test
```suggestion
        raise Exception(f"Test failed with error: {str(e)}")
```

**Review by OjusWiZard — CHANGES_REQUESTED** [review]
Beautifully done @Mohsinsiddi! I've asked somethings. Also, let's organise the dev/testing dependencies with the current method (poetry) only. It will keep things clean and modular.

File: tests/test_run_service.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    log_file = Path(f'test_run_service_{timestamp}.log')
+    logger = setup_logging(log_file)
+    
+    logger.info("Starting run_service.py test")
+    logger.info(f"Log file: logs/{log_file}")
+
+    
+    try:
+        script_path = Path("./run_service.sh")
+        os.chmod(script_path, 0o755)
```
[the remark this answers is not in the document]
Comment (Mohsinsiddi) [reply tests/test_run_service.py:196]: removed it

File: tests/test_run_service.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+                
+            except pexpect.EOF:
+                logger.info("Process completed")
+                break
+            except pexpect.TIMEOUT:
+                logger.error("Timeout waiting for prompt")
+                break
+            except Exception as e:
+                logger.error(f"Error handling prompt: {str(e)}")
+                break
```
[the remark this answers is not in the document]
Comment (Mohsinsiddi) [reply tests/test_run_service.py:241]: added raise

**Pull request closed without merging**